File: frontend/src/types/translation-report.ts
export interface TranslationReport {
  id: string;
  clinic_id: string;
  message_id: string | null;
  reported_by: string;
  source_language: string;
  target_language: string;
  original_text: string;
  translated_text: string;
  corrected_text: string | null;
  error_type: string;
  severity: string;
  medical_term_id: string | null;
  status: string;
  reviewer_id: string | null;
  reviewer_notes: string | null;
  reviewed_at: string | null;
  created_at: string;
}

export interface TranslationQAStats {
  total_reports: number;
  pending_count: number;
  resolved_count: number;
  critical_count: number;
  by_error_type: Record<string, number>;
  by_language_pair: { source: string; target: string; count: number }[];
  accuracy_score: number | null;
}

export interface TranslationReportPaginated {
  items: TranslationReport[];
  total: number;
  limit: number;
  offset: number;
}

File: backend/app/schemas/translation_report.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class TranslationReportCreate(BaseModel):
    message_id: uuid.UUID | None = None
    source_language: str
    target_language: str
    original_text: str
    translated_text: str
    corrected_text: str | None = None
    error_type: str  # wrong_term / wrong_meaning / awkward / omission / other
    severity: str  # critical / minor
    medical_term_id: uuid.UUID | None = None


class TranslationReportResponse(BaseModel):
    id: uuid.UUID
    clinic_id: uuid.UUID
    message_id: uuid.UUID | None
    reported_by: uuid.UUID
    source_language: str
    target_language: str
    original_text: str
    translated_text: str
    corrected_text: str | None
    error_type: str
    severity: str
    medical_term_id: uuid.UUID | None
    status: str
    reviewer_id: uuid.UUID | None
    reviewer_notes: str | None
    reviewed_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class TranslationReportReview(BaseModel):
    status: str  # reviewed / resolved / dismissed
    reviewer_notes: str | None = None
    corrected_text: str | None = None


class TranslationQAStats(BaseModel):
    total_reports: int
    pending_count: int
    resolved_count: int
    critical_count: int
    by_error_type: dict[str, int]
    by_language_pair: list[dict]
    accuracy_score: float | None  # percentage of messages without issues
